core/src/main/java/org/keycloak/jose/jws/crypto/HashUtils.java

```diff
@@ -30,15 +32,20 @@
  */
 public class HashUtils {
 
-    // See "at_hash" and "c_hash" in OIDC specification
-    public static String oidcHash(String jwtAlgorithmName, String input) {
+    // See:
+    // - "at_hash" and "c_hash" in OIDC specification (full = false)
+    // - "ath" in DPoP specification (full = true)
```

Comment:
Basically, this utility codes are used for a client and a resource server, not for an authorization server (keycloak) because these can be used for creating and verifying DPoP proof that is sent from a client to a resource server.

However, keycloak's UserInfo endpoint can receive an access token in DPoP. Are these codes used for?

Reply:
Exactly - the second ("ath") variant is meant to be used in UserInfo. We're don't have DPoP support for UserInfo yet as we have to ensure its standard-compliance first (see #8801).

Also, the second variant might be used by Java adapters in the future, to implement DPoP-compliant resource servers.